packages/cli/src/controllers/tags.controller.ts

```diff
@@ -29,61 +26,48 @@ export class TagsController {
 		next();
 	}
 
-	// Retrieves all tags, with or without usage count
 	@Get('/')
-	async getAll(req: TagsRequest.GetAll): Promise<TagEntity[] | ITagWithCountDb[]> {
-		const { withUsageCount } = req.query;
-		if (withUsageCount === 'true') {
-			return this.tagsRepository
-				.find({
-					select: ['id', 'name', 'createdAt', 'updatedAt'],
-					relations: ['workflowMappings'],
-				})
-				.then((tags) =>
-					tags.map(({ workflowMappings, ...rest }) => ({
-						...rest,
-						usageCount: workflowMappings.length,
-					})),
-				);
-		}
-
-		return this.tagsRepository.find({ select: ['id', 'name', 'createdAt', 'updatedAt'] });
+	async getAll(req: TagsRequest.GetAll) {
+		return this.tagService.getAll({ withUsageCount: req.query.withUsageCount === 'true' });
 	}
 
-	// Creates a tag
 	@Post('/')
-	async createTag(req: TagsRequest.Create): Promise<TagEntity> {
-		const newTag = this.tagsRepository.create({ name: req.body.name.trim() });
+	async createTag(req: TagsRequest.Create) {
+		const tag = this.tagService.toEntity({ name: req.body.name });
 
-		await this.externalHooks.run('tag.beforeCreate', [newTag]);
-		await validateEntity(newTag);
+		await this.externalHooks.run('tag.beforeCreate', [tag]);
```

Comment:
we should move all the `externalHooks`  calls to the service as well. eventually all the controllers should only do input validation and all business logic should go into services.